matter_server/server/device_controller.py

```diff
@@ -630,59 +658,82 @@ async def _subscribe_node(self, node_id: int) -> None:
+            # device is online/offline while the second interval actually subscribes to
+            # the attributes and/or events.
+            base_sub = await self._setup_subscription(
+                node,
+                attr_subscriptions=list(BASE_SUBSCRIBE_ATTRIBUTES),
+                interval_floor=interval_floor,
+                interval_ceiling=interval_ceiling,
+                # subscribe to urgent device events only (e.g. button press etc.) only
+                event_subscriptions=[
                     Attribute.EventPath(
                         EndpointId=None, Cluster=None, Event=None, Urgent=1
                     )
                 ],
-                returnClusterObject=False,
-                subscriptionParameters=Attribute.SubscriptionParameters(
-                    interval_floor, interval_ceiling
-                ),
-                # Use fabricfiltered as False to detect changes made by other controllers
-                # and to be able to provide a list of all fabrics attached to the device
-                fabricFiltered=False,
-                autoResubscribe=True,
-            ).raise_on_error()
-            sub: Attribute.SubscriptionTransaction = await future
+            )
+            attr_sub = await self._setup_subscription(
+                node,
+                attr_subscriptions=attr_subscriptions,
+                interval_floor=interval_floor,
+                interval_ceiling=interval_ceiling,
+            )
+        # if we reach this point, it means the node could be resolved
+        # and the initial subscription succeeded, mark the node available.
+        self._subscriptions[node_id] = (base_sub, attr_sub)
+        node.available = True
+        # update attributes with current state from read request
+        # NOTE: Make public method upstream for retrieving the attributeTLVCache
+        # pylint: disable=protected-access
+        for sub in (base_sub, attr_sub):
+            tlv_attributes = sub._readTransaction._cache.attributeTLVCache
+            node.attributes.update(parse_attributes_from_read_result(tlv_attributes))
+        node_logger.info("Subscription succeeded")
+        self.server.signal_event(EventType.NODE_UPDATED, node)
+
+    async def _setup_subscription(
+        self,
+        node: MatterNodeData,
+        attr_subscriptions: list[Attribute.AttributePath],
+        interval_floor: int = 0,
+        interval_ceiling: int = 60,
+        event_subscriptions: list[Attribute.EventPath] | None = None,
+    ) -> Attribute.SubscriptionTransaction:
+        """Handle Setup of a single Node AttributePath(s) subscription."""
+        node_id = node.node_id
+        node_logger = LOGGER.getChild(f"[node {node_id}]")
+        assert self.chip_controller is not None
+        node_logger.debug("Setting up attributes and events subscription.")
+        self._last_subscription_attempt[node_id] = 0
+        assert self.server.loop is not None
```

Comment:
We're using the loop attribute multiple times. I'd store it in a local variable first.